### PR TITLE
Support syft

### PR DESCRIPTION
**Pull request by Daniel-GrunbergerCA**
## PR Type:
Enhancement

___
## PR Description:
This PR introduces support for handling Software Bill of Materials (SBOMs) in Syft format. The changes are as follows:
- Replaces the types of SPDX SBOMs with Syft ones in the watcher.
- Updates the watcher tests to reflect these changes.
- Fixes dependencies to the latest Storage version.

___
## PR Main Files Walkthrough:
<details> <summary>files:</summary>

- `watcher/watcher_test.go`: Replaced SPDX SBOMs with Syft SBOMs in the test cases. Updated the expected SBOM names in the tests.
- `watcher/watcher.go`: Replaced SPDX SBOMs with Syft SBOMs in the SBOM event handlers. Updated the SBOM watcher to handle Syft SBOMs.
- `go.sum`: Updated the project dependencies.
- `go.mod`: Updated the project dependencies.
</details>

___
## User Description:
This PR introduces support for watching `sboms` and `filteredsboms` on `syft` format. It replaces the types of `spdx` sboms with `syft` ones.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/kubescape/operator
 
-go 1.20
+go 1.21.3
```
[the remark this answers is not in the document]
Comment (matthyx) [reply go.mod:3]: maybe we need to update the test runners to 1.21?

**Comment by matthyx** [comment]
> * Go linting: failure

Can you check why?

**Comment by Daniel-GrunbergerCA** [comment]
> > * Go linting: failure
> 
> Can you check why?

It's because I don't check the error returned:
`Error return value of `(github.com/kubescape/storage/pkg/generated/clientset/versioned/typed/softwarecomposition/v1beta1.SBOMSPDXv2p3Interface).Delete` is not checked (errcheck)`

I don't check since it is deleting older spdx sboms, and we can't do anything it there is an error (also I have asked @vladklokun if he thinks we should leave this check)
There is also another [place](https://github.com/kubescape/operator/blob/main/watcher/watcher.go#L274) on the code where we didn't check before and I also don't check ..

File: watcher/watcher.go [hunk 2]
```diff
@@ -327,7 +327,7 @@ func (wh *WatchHandler) HandleSBOMEvents(sbomEvents <-chan watch.Event, errorCh
 	defer close(errorCh)
 
 	for event := range sbomEvents {
-		obj, ok := event.Object.(*spdxv1beta1.SBOMSummary)
+		obj, ok := event.Object.(*spdxv1beta1.SBOMSyft)
```
Comment (dwertent) [inline watcher/watcher.go:330]: @Daniel-GrunbergerCA Are we not using summary objects?

**Pull request merged**